lib/kubernetes-deploy/runner_task.rb

```diff
@@ -15,43 +15,115 @@ def initialize(task_template, namespace, context)
     end
 
-    def initialize(namespace:, context:, logger:)
+    def initialize(namespace:, context:, logger:, max_watch_seconds: nil)
       @logger = logger
       @namespace = namespace
       @kubeclient = build_v1_kubeclient(context)
       @context = context
+      @max_watch_seconds = max_watch_seconds
     end
 
     def run(*args)
       run!(*args)
       true
+    rescue DeploymentTimeoutError
+      false
     rescue FatalDeploymentError => error
-      @logger.fatal "#{error.class}: #{error.message}"
+      if error.message != error.class.to_s
+        @logger.summary.add_action(error.message)
+        @logger.print_summary(:failure)
+      end
       false
     end
 
-    def run!(task_template:, entrypoint:, args:, env_vars: [])
+    def run!(task_template:, entrypoint:, args:, env_vars: [], verify_result: true)
       @logger.reset
-      @logger.phase_heading("Validating configuration")
+      @logger.phase_heading("Initializing deploy")
       validate_configuration(task_template, args)
       if kubectl.server_version < Gem::Version.new(MIN_KUBE_VERSION)
         @logger.warn(KubernetesDeploy::Errors.server_version_warning(kubectl.server_version))
       end
-      @logger.phase_heading("Fetching task template")
       raw_template = get_template(task_template)
 
-      @logger.phase_heading("Constructing final pod specification")
       rendered_template = build_pod_template(raw_template, entrypoint, args, env_vars)
-
-      validate_pod_spec(rendered_template)
-
+      validate_restart_policy(rendered_template, verify_result)
+      pod = Pod.new(namespace: @namespace, context: @context, logger: @logger, log_on_success: false,
+                    definition: rendered_template.to_hash.deep_stringify_keys, statsd_tags: [])
+      pod.validate_definition(kubectl)
+      @logger.info("Configuration valid")
       @logger.phase_heading("Creating pod")
       @logger.info("Starting task runner pod: '#{rendered_template.metadata.name}'")
+
+      pod_creation_time = Time.now.utc
+      pod.deploy_started_at = pod_creation_time
       @kubeclient.create_pod(rendered_template)
+
+      if verify_result
+        sync_mediator = SyncMediator.new(namespace: @namespace, context: @context, logger: @logger)
+        watch_and_report(pod, pod_creation_time, sync_mediator)
+      else
+        warning = <<~MSG
+          Result verification is disabled for this task.
+          This means the desired pod was successfully created, but the runner did not make sure it actually succeeded.
+        MSG
+        @logger.summary.add_paragraph(ColorizedString.new(warning).yellow)
+      end
     end
 
     private
 
+    def watch_and_report(pod, deploy_started_at, sync_mediator)
+      sync_interval = 5.seconds
+      last_loop_time = last_log_time = deploy_started_at
+      @logger.info("Logs from #{pod.id}:")
+
+      loop do
+        pod.sync(sync_mediator)
+        logs = pod.fetch_logs(kubectl, since: last_log_time.to_datetime.rfc3339)
```

Comment:
nitpick but it feels weird to me that the caller is responsible for the `to_datetime.rfc3339` stuff here. Shouldn't the `fetch_logs` method do that itself?